### ISSUE TITLE
Customer assigned to wrong server - stuck at test case 3 of project

### ISSUE THREAD

**Issue by bzastudent293** [nus-cs2030/2324-s2#282]
# Summary
Hi, was anyone stuck at test case 3 of the project where the customer is being assigned to the wrong server for the wait event, I am struggling to get back this test case? Any help will be really appreciated.

## Description 
Describe the problem and code snippets if necessary 

## Screenshots (if any): 
Insert Images here if necessary
<img width="462" alt="Screenshot 2024-03-21 at 6 03 33 PM" src="https://github.com/nus-cs2030/2324-s2/assets/134027838/654abc39-f86e-43c9-95fe-9602ce877f38">

**Comment by iceflakes12** [comment]
Hihi, I tilted so hard at this exact same problem also, so I hope that what I write below will be able to save your time.

Let me give you an example to help you understand where you might have gone wrong.
Let's say server 1 has customers 2 and 3 in its queue, and is currently serving customer 1. Let's put it all in a list [1, 2, 3]. We can all agree that queuelength is 2.

Let's say server 1 has finished serving customer 1, and the Done event is being executed right now. What I suspect your current implementation does is just remove customer 1 from the list, and then this makes the list become [2, 3]. However, what happens is that when you do this, your queuelength becomes 1. This would typically work for most circumstances, but things change when rest comes into play. While the server is resting, your queuelength needs to remain as 2, as identified in the specs. Only after your server has finished resting, would your queuelength become 1.

How I got around this was instead of removing customer 1 from the list during the Done event, I have a Rest event that removes customer 1 from the list only after server has finished resting.
